### DISCUSSION TITLE
DrawingModelInterpolator example

### DISCUSSION THREAD

**Discussion by Curwe** [patrykandpatrick/vico#470] [Questions] [answered]
Hello guys.
It would be awesome to get a DrawingModelInterpolator example in the sample app. If I understand the code correct it would allow me to change the color of a column on the fly? Eg, the column with the marker over it would be one color and the rest would be a different color? But It's a bit unclear on how it works.

**Comment by patrickmichalik** [comment, marked as answer]
Hello! `DrawingModelInterpolator` animates `DrawingModel`s, which don’t store color information. An example implementation is [`DefaultDrawingModelInterpolator`](https://github.com/patrykandpatrick/vico/blob/master/vico/core/src/main/java/com/patrykandpatrick/vico/core/entry/diff/DefaultDrawingModelInterpolator.kt). What you’ve described isn’t possible yet, but we’ll be looking into adding such functionality.

**Reply by Curwe** [reply]
@patrickmichalik thanks for the answer.  

While I waited I forked the lib and added the color switching there.
It's a custom FloatEntry that take a LineComponent. And in core/ ColumnChartI've added this 

```
model.entries.forEachIndexed { index, entryCollection ->
            
            //column = columns.getRepeating(index) Set column per entry 
            drawingStart = getDrawingStart(index, model.entries.size) - horizontalScroll

            entryCollection.forEachInAbsolutelyIndexed(chartValues.minX..chartValues.maxX) { entryIndex, entry ->

                column = columns.getRepeating(index)
                if (entry is FloatColoredEntry) column = entry.lineComponent
                
                ....
}
```
 
This doesn't break anything with the colors in the sample app, and allows me to change the colors on the go. 
So each entry can hold a lineComponent. And then when I tap a column I pass the index (x) to my view model and just change the replace the entry with a new color. 
You could still pass columns in the columnChart with this.  

Maybe this could help you guys a bit with this.

**Reply by patrickmichalik** [reply]
Thanks for sharing, @Curwe! While such an approach has the benefit of being quite straightforward, there are a few issues due to which it’s not entirely viable. The main one is that chart data can be handled asynchronously, but some visual changes must be instant. When dark mode is toggled, for example, a chart’s colors should change immediately, but a data update run via `CartesianChartModelProducer` doesn’t block the main thread. Thus, our current plan is to introduce a callback-based mutation system. This solution will be compatible with asynchronous data updates and should also use a bit less memory.
